neuraxle/steps/misc.py

```diff
@@ -223,6 +221,10 @@ def get_name_tape(self) -> List[str]:
         """
         return self.name_tape
 
+    def reset(self):
+        self.data = []
+        self.name_tape = []
```

Comment:
BaseStep.reset: 
```
    def reset(self):
        self.teardown()
        self.initialize()
```

and def initialize & teardown here.

Reply:
this way a whole pipeline can reset to virgin anytime.

Reply:
a TapeCallbackFunction is not a step.